### PR TITLE
Add jumping cheat and prevent invincible from allowing the health bar to go in red

### PR DESCRIPTION
**Pull request by grasmanek94**

### CODE REVIEW

File: cheats.js [hunk 1]
```diff
@@ -242,6 +326,99 @@ ig.module("cheats").requires("game.feature.player.player-level", "game.feature.p
 			return this.parent(...args);
 		},
 	});
+	var a = Vec2.create(), d = {};
```
Comment (ZeikJT) [inline cheats.js:329]: These variable names a,d are too generic and don't seem to be used below?

File: cheats.js [hunk 2]
```diff
@@ -242,6 +326,99 @@ ig.module("cheats").requires("game.feature.player.player-level", "game.feature.p
...
+
+            setCheatValue("jumphighermodifier", old_value);
+
+			cheat_perform_jump = false;
+            return this.parent(a,...args);
+        },
+		doFloatJump(...args) {
+			cheat_perform_jump = false;
+		    return this.parent(...args);
+		},
```
Comment (ZeikJT) [inline cheats.js:386]: Indentation here seems to be all over the place.

File: cheats.js [hunk 3]
```diff
@@ -24,7 +24,18 @@ const CHEAT_CONFIG = [
 	["enemydropcheat",          {defaultValue: true, type: "CHECKBOX"}],
 	["plantdropcheat",          {defaultValue: true, type: "CHECKBOX"}],
 	["donotremovetrophypoints", {defaultValue: true, type: "CHECKBOX", preconditions: ["NEW_GAME_PLUS"]}],
+	["jumphigher",              {defaultValue: true, type: "CHECKBOX"}],
+	["jumphighermodifier",      {defaultValue: 5,    type: "SLIDER", min: 1, max: 10,  requires: ["jumphigher"]}],
+	["jumpfurther",             {defaultValue: 10,   type: "SLIDER", min: 10, max: 40,requires: ["jumphigher"]}],
```
Comment (ZeikJT) [inline cheats.js:29]: Missing a space between the comma and requires.

File: cheats.js [hunk 4]
```diff
@@ -24,7 +24,18 @@ const CHEAT_CONFIG = [
 	["enemydropcheat",          {defaultValue: true, type: "CHECKBOX"}],
 	["plantdropcheat",          {defaultValue: true, type: "CHECKBOX"}],
 	["donotremovetrophypoints", {defaultValue: true, type: "CHECKBOX", preconditions: ["NEW_GAME_PLUS"]}],
+	["jumphigher",              {defaultValue: true, type: "CHECKBOX"}],
+	["jumphighermodifier",      {defaultValue: 5,    type: "SLIDER", min: 1, max: 10,  requires: ["jumphigher"]}],
+	["jumpfurther",             {defaultValue: 10,   type: "SLIDER", min: 10, max: 40,requires: ["jumphigher"]}],
+	["skipintro",               {defaultValue: true, type: "CHECKBOX"}],
+	["unlimiteddashes",         {defaultValue: true, type: "CHECKBOX"}],
+	["runspeed",                {defaultValue: true, type: "CHECKBOX"}],
+	["runspeedmultiplier",      {defaultValue: 10,   type: "SLIDER", min: 1, max: 100,requires: ["runspeed"]}],
```
Comment (ZeikJT) [inline cheats.js:33]: Same as above, missing a space between the comma and requires.

**Review by ZeikJT — CHANGES_REQUESTED** [review]
Overall these seem like really cool additions! Thanks for working on adding them :)

**Comment by grasmanek94** [comment]
> Overall these seem like really cool additions! Thanks for working on adding them :)

Yeah most of these I had already modded myself into the compiled game.js, I ported them here :)

**Pull request merged**